src/Microsoft.ML.Tokenizers/PreTokenizer/RegexPreTokenizer.cs

```diff
@@ -35,6 +35,8 @@ public RegexPreTokenizer(Regex regex, IReadOnlyDictionary<string, int>? specialT
 
             if (specialTokensEncoder is { Count: > 0 })
             {
+                // We create this Regex object without a timeout because we know it will perform the match operation in O(N) time complexity.
```

Comment:
It's actually O(N * NumberOfKeys), but if the latter is a constant, then it's still O(N).

Reply:
Yes, NumberOfKeys is constant after creating the pre-tokenizer. I added extra sentence clarifying that.